config/src/test/kotlin/org/springframework/security/config/annotation/web/AuthorizeHttpRequestsDslTests.kt

```diff
@@ -961,4 +964,63 @@ class AuthorizeHttpRequestsDslTests {
         }
 
     }
+
+    @Test
+    fun `request when fully authenticated configured then responds ok`() {
+        this.spring.register(FullyAuthenticatedConfig::class.java).autowire()
+
+        this.mockMvc.post("/path") {
+            with(SecurityMockMvcRequestPostProcessors.user("user").roles("USER"))
+            with(csrf())
+        }.andExpect {
+            status {
+                isOk()
+            }
+        }
+    }
+
+    @Test
+    fun `request when fully authenticated configured and remember-me token then responds unauthorized`() {
+        this.spring.register(FullyAuthenticatedConfig::class.java).autowire()
+        val rememberMe = RememberMeAuthenticationToken("key", "user",
+                AuthorityUtils.createAuthorityList("ROLE_USER"))
+
+        this.mockMvc.post("/path") {
+            with(SecurityMockMvcRequestPostProcessors.user("user").roles("USER"))
```

Comment:
Something doesn't quite make sense about this test. There are two `with` declarations setting the authentication, when there should only be one. Also, it is returning `isUnauthorized` when it should be `isForbidden`. Can you look into this please?

Reply:
Fixed.